### PR TITLE
Adding Squeeze net in Ivy 

### PR DESCRIPTION
**Pull request by abdulasiraj**

### CODE REVIEW

**Comment by abdulasiraj** [comment]
There's a problem in test `Failed: [undefined]ValueError: negative dimensions are not allowed`. I've used Alexnet as template but it's creating issue on Squeeze Net.

File: ivy_models/squeezenet/squeezenet.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
+from ivy_models.helpers import load_torch_weights
+import ivy
+import builtins
+
+class Fire(ivy.Module):
```
Comment (juliagsy) [inline ivy_models/squeezenet/squeezenet.py:5]: please rename this to SqueezeNetFire for convention

File: ivy_models/squeezenet/squeezenet.py [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+    new_mapping = new_key
+    if "weight" in old_key:
+        new_mapping = {"key_chain": new_key, "pattern": "b c h w -> w h c b"}
+    elif "bias" in old_key:
+        new_mapping = {"key_chain": new_key, "pattern": "h -> 1 h 1 1"}
+
+    return new_mapping
+
+
+def squeezeNet(version: str = "1_0", num_classes: int = 1000, dropout: float = 0.5, v=None, pretrained=True):
```
Comment (juliagsy) [inline ivy_models/squeezenet/squeezenet.py:84]: please keep them all lower case, ie, squeezenet

File: ivy_models/squeezenet/squeezenet.py [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+    def _forward(self, x):
+        x = self.features(x)
+        x = self.classifier(x)
+        return ivy.flatten(x, start_dim=1)
+    
+
+def _squeezenet_torch_weights_mapping(old_key, new_key):
+    new_mapping = new_key
+    if "weight" in old_key:
+        new_mapping = {"key_chain": new_key, "pattern": "b c h w -> w h c b"}
```
Comment (juliagsy) [inline ivy_models/squeezenet/squeezenet.py:77]: `... -> h w c b` is better just in case the dimension is not a square

**Review by juliagsy — CHANGES_REQUESTED** [review]
Thanks for the great work! Just some naming conventions to fix, then we should be able to get it merged, thanks!

**Pull request merged**